### PR TITLE
Define reflection for glm types

### PR DESCRIPTION
**Pull request by luishfonseca**
# Description

Implement Constructible and Fields traits for all glm types

## Checklist

- [x] Self-review changes.
- [x] Evaluate impact on the documentation.
- [x] Ensure test coverage.

### CODE REVIEW

File: .githooks/pre-commit
Comment (RiscadoA) [inline .githooks/pre-commit:1]: wtf but sure

File: core/include/cubos/core/reflection/external/glm.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+CUBOS_REFLECT_EXTERNAL_DECL(glm::vec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::bvec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dvec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::ivec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::uvec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::vec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dvec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::bvec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::ivec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::uvec4);
```
Comment (RiscadoA) [inline core/include/cubos/core/reflection/external/glm.hpp:45]: Some spacing would be appreciated between types of different 'groups' (mats vs vecs etc)

File: core/src/cubos/core/reflection/external/glm.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    uintptr_t get(const void* instance) const override
+    {
+        return reinterpret_cast<uintptr_t>(&(static_cast<const T*>(instance)[mColumn]));
+    }
+
+private:
+    size_t mColumn;
+};
```
Comment (RiscadoA) [inline core/src/cubos/core/reflection/external/glm.cpp:31]: Didn't think of doing it this way, but its cool.

**Review by RiscadoA — CHANGES_REQUESTED** [review]

File: .githooks/pre-commit
Comment (luishfonseca) [reply .githooks/pre-commit:1]: :sweat_smile:

**Review by RiscadoA — APPROVED** [review]

**Pull request merged**